modules/installation-vsphere-installer-infra-requirements.adoc

```diff
@@ -376,6 +376,343 @@ Additionally, the user requires some `ReadOnly` permissions, and some of the rol
+`Network.Assign`
+`Read Only`
+
+|Virtual Machine Folder
+|Always
+|
+[%hardbreaks]
+`Read Only`
+`VirtualMachine.Config.AddExistingDisk`
+`VirtualMachine.Config.AddRemoveDevice`
+`VirtualMachine.Config.AdvancedConfig`
+`VirtualMachine.Config.Annotation`
+`VirtualMachine.Config.CPUCount`
+`VirtualMachine.Config.DiskExtend`
+`VirtualMachine.Config.Memory`
+`VirtualMachine.Config.Settings`
+`VirtualMachine.Interact.PowerOff`
+`VirtualMachine.Interact.PowerOn`
+`VirtualMachine.Inventory.CreateFromExisting`
+`VirtualMachine.Inventory.Delete`
+`VirtualMachine.Provisioning.Clone`
+`VirtualMachine.Provisioning.DeployTemplate`
+
+|vSphere vCenter Datacenter
+|If the installation program creates the virtual machine folder. For UPI, VirtualMachine.Inventory.Create and VirtualMachine.Inventory.Delete privileges are optional if your cluster does not use the Machine API.
+|
+[%hardbreaks]
+`Resource.AssignVMToPool`
+`Read Only`
+`VirtualMachine.Config.AddExistingDisk`
+`VirtualMachine.Config.AddRemoveDevice`
+`VirtualMachine.Interact.PowerOff`
+`VirtualMachine.Interact.PowerOn`
+`VirtualMachine.Provisioning.DeployTemplate`
+|===
+====
+
+.Minimum permissions for the machine API
+[%collapsible]
+====
+[cols="4a,4a,3a",options="header"]
+|===
+|vSphere object for role
+|When required
+|Required privileges
+
+|vSphere vCenter
+|Always
+|
+[%hardbreaks]
+`InventoryService.Tagging.AttachTag`
+`InventoryService.Tagging.CreateCategory`
+`InventoryService.Tagging.CreateTag`
+`InventoryService.Tagging.DeleteCategory`
+`InventoryService.Tagging.DeleteTag`
+`InventoryService.Tagging.EditCategory`
+`InventoryService.Tagging.EditTag`
+`Sessions.ValidateSession`
+`Read Only`
+`StorageProfile.Update`
+`StorageProfile.View`
+
+|vSphere vCenter Cluster
+|If you intend to create VMs in the cluster root
+|
+[%hardbreaks]
+`Resource.AssignVMToPool`
+`Read Only`
+
+|vSphere vCenter Resource Pool
+|If you provide an existing resource pool in the `install-config.yaml` file
+|
+[%hardbreaks]
+`Read Only`
+
+|vSphere Datastore
+|Always
+|
+[%hardbreaks]
+`Datastore.AllocateSpace`
+`Datastore.Browse`
+`Read Only`
+
+|vSphere Port Group
+|Always
+|
+[%hardbreaks]
+`Network.Assign`
+`Read Only`
+
+|Virtual Machine Folder
+|Always
+|
+[%hardbreaks]
+`Read Only`
+`VirtualMachine.Config.AddRemoveDevice`
+`VirtualMachine.Config.AdvancedConfig`
+`VirtualMachine.Config.Annotation`
+`VirtualMachine.Config.CPUCount`
+`VirtualMachine.Config.DiskExtend`
+`VirtualMachine.Config.Memory`
+`VirtualMachine.Config.Settings`
+`VirtualMachine.Interact.PowerOff`
+`VirtualMachine.Interact.PowerOn`
+`VirtualMachine.Inventory.CreateFromExisting`
+`VirtualMachine.Inventory.Delete`
+`VirtualMachine.Provisioning.Clone`
+`VirtualMachine.Provisioning.DeployTemplate`
+
+|vSphere vCenter Datacenter
+|If the installation program creates the virtual machine folder. For UPI, VirtualMachine.Inventory.Create and VirtualMachine.Inventory.Delete privileges are optional if your cluster does not use the Machine API.
```

Comment:
This section is irrelivant as it goes against the intent of the API machine table?

Reply:
i think keeping this is fine, but maybe link to the machine API privileges section

Reply:
Updated for numerous tables.